### PR TITLE
Add menu item to workspace

### PR DESCRIPTION
**Pull request by rschwanhold**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,13 +2,11 @@ language: smalltalk
...
 
 before_install:
-  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then ls -la ; fi
-  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then pwd ; fi
-  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then ldd hunspell.so ; fi
-  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then nm -D hunspell.so  ; fi
   - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then sudo apt-get install libhunspell-1.3-0:i386          ; fi
   - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then ln -s /usr/lib/i386-linux-gnu/libhunspell-1.3.so.0 libhunspell-1.3.so.0 ; fi
-  
+  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then ls -la ; fi
```
Comment (rschwanhold) [inline .travis.yml]: the travis file shouldn't be change. I need to undo this before merging

File: packages/Spellcheck-Core.package/SpellcheckCorrectionUI.class/instance/selectedAlternativeIndex..st [hunk 2]
```diff
@@ -1,8 +1,9 @@
...
-	Transcript show: inputText.
-	self next.
-	"selectedIndex := anIndex."
-	"self changed: #selectedAlternativeIndex."
\ No newline at end of file
+	selectedIndex := anIndex.
+	self changed: #selectedAlternativeIndex.
+	workspace 
+		ifNotNil: [self replaceWorkspaceTextWithAlternative.] 
+		ifNil: [inputText replaceFrom: currentKey to: (wrongWords at: currentKey) with: (self alternatives at: anIndex).].
```
Comment (rschwanhold) [inline packages/Spellcheck-Core.package/SpellcheckCorrectionUI.class/instance/selectedAlternativeIndex..st]: I am not sure if we should restrict the tool to only be used in the workspace!? We could get rid of the case `ifNil`

File: packages/Spellcheck-Core.package/Workspace.extension/instance/addOriginalModelItemsToWindowMenu..st [hunk 3]
```diff
@@ -0,0 +1,26 @@
+*Spellcheck-Core
+addOriginalModelItemsToWindowMenu: aMenu
```
Comment (rschwanhold) [inline packages/Spellcheck-Core.package/Workspace.extension/instance/addOriginalModelItemsToWindowMenu..st]: This is the menu that was created previously

File: packages/Spellcheck-Tests.package/FFITests.class/instance/ffiGetString.st [hunk 4]
```diff
@@ -0,0 +1,4 @@
+testing
```
Comment (rschwanhold) [inline packages/Spellcheck-Tests.package/FFITests.class/instance/ffiGetString.st]: This also shouldn't have changed

File: packages/Spellcheck-Tests.package/FFITests.class/instance/testDirectLibraryAccess.st [hunk 5]
```diff
@@ -0,0 +1,6 @@
+testing
```
Comment (rschwanhold) [inline packages/Spellcheck-Tests.package/FFITests.class/instance/testDirectLibraryAccess.st]: This neither

File: packages/Spellcheck-Core.package/SpellcheckCorrectionUI.class/instance/input..st [hunk 6]
```diff
@@ -0,0 +1,4 @@
+as yet unclassified
+input: aWorkspace
```
Comment (rschwanhold) [inline packages/Spellcheck-Core.package/SpellcheckCorrectionUI.class/instance/input..st]: maybe we should change the name of the method...?

**Comment by rschwanhold** [comment]
It is not working perfectly yet but the main functionality is there.
There are a few things that I don't like:
1. The way we change the content of the workspace: we are resetting the whole content instead of only changing the wrong words (it works fine but it is a bit ugly)
2. the order of the words is random and I am not sure if we can sort them...
3. Sometimes an error appears: This is the case because if we change the word `goood` to `good` in a sentence, then the start and end indexes of the other words are 'off by 1'. This causes sometime an `Out-of-bounds` -error. (this needs to be fixed)
4. I haven't wrote any tests yet

**Comment by youri-k** [comment]
My ideas concerning your comment:
1. I think it's the only possible way to do it like that, since the textMorph (IMO rightfully) doesn't have an interface to change single words
2. I think this can be solved if we handle the keys of the dictionary as SortedCollection since these represent the start points of wrong words
3. true
4. also true and important
5. nice work

**Pull request merged**